server/game/cards/locations/07/abandonedstronghold.js

```diff
@@ -0,0 +1,40 @@
+const DrawCard = require('../../../drawcard.js');
+
+class AbandonedStronghold extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Give character +STR',
+            condition: () => this.game.currentChallenge,
+            cost: ability.costs.kneelSelf(),
+            target: {
+                activePromptTitle: 'Select a character',
+                cardCondition: card => (
+                    card.location === 'play area' && 
+                    card.getType() === 'character' && 
+                    this.game.currentChallenge.isDefending(card))
+            },
+            handler: context => {
+                let strBoost = this.getNumberOfBuilders();
+                this.untilEndOfChallenge(ability => ({
+                    match: context.target,
+                    effect: ability.effects.modifyStrength(strBoost)
+                }));
+
+                this.game.addMessage('{0} kneels {1} to give {2} +{3} STR until the end of the challenge', 
+                                      context.player, this, context.target, strBoost);
+            }
+        });
+    }
+
+    getNumberOfBuilders() {
+        let cards = this.controller.filterCardsInPlay(card => {
```

Comment:
FYI, you could just return `this.controller.getNumberOfCardsInPlay(card => ...)` instead, but this is less awkward than the `any` checks, so I'll let it through for now.